src/main/java/net/gegy1000/plasmid/block/CustomBlock.java

```diff
@@ -0,0 +1,92 @@
+package net.gegy1000.plasmid.block;
+
+import com.google.common.base.Preconditions;
+import net.gegy1000.plasmid.registry.TinyRegistry;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.BlockPos;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class CustomBlock {
+    private static final TinyRegistry<CustomBlock> REGISTRY = TinyRegistry.newStable();
+    private static final Map<BlockPos, CustomBlock> BLOCK_MAP = new HashMap<>();
```

Comment:
1. A Long2ObjectOpenHashMap will boost the speed of the map here, as the block positions can be stored as long values.
2. A way to clear the map would be needed to prevent memory leaking.

Reply:
Ive always wondered how much faster FastUtil maps are. how fast are they compared to normal java maps?

Reply:
They're much faster than normal maps and should always be used if possible.